ssl/ssl_locl.h

```diff
@@ -1314,7 +1314,12 @@ typedef struct ssl3_state_st {
      * that the server selected once the ServerHello has been processed.
      */
     unsigned char *alpn_selected;
-    unsigned alpn_selected_len;
+    unsigned int alpn_selected_len;
+    /* used by the server to know what options were proposed */
+    unsigned char *alpn_proposed;
+    size_t alpn_proposed_len; /* size_t for PACKET functions */
```

Comment:
Nit, comment is odd: this isn't just for "PACKET functions", this is the length of the allocated buffer.